.circleci/config.yml

```diff
@@ -149,9 +149,14 @@ commands:
                   DD_TRACE_AGENT_URL: << parameters.trace_agent_url >> 
                   RIOT_RUN_RECOMPILE_REQS: "<< pipeline.parameters.riot_run_latest >>"
                 command: |
-                  # Sort the hashes to ensure a consistent ordering/division between each node
-                  riot list --hash-only '<<parameters.pattern>>' | sort | circleci tests split | xargs -n 1 -I {} ./scripts/ddtest riot -v run --exitfirst --pass-env -s {} $([[ << pipeline.parameters.coverage >> == false ]] && echo '--no-cov' )
-                  ./scripts/check-diff ".riot/requirements/" "Changes detected after running riot. Consider deleting changed files, running scripts/compile-and-prune-test-requirements and committing the result."
+                  if scripts/needs_testrun.py -v "$CIRCLE_JOB"
+                  then
+                    # Sort the hashes to ensure a consistent ordering/division between each node
+                    riot list --hash-only '<<parameters.pattern>>' | sort | circleci tests split | xargs -n 1 -I {} ./scripts/ddtest riot -v run --exitfirst --pass-env -s {} $([[ << pipeline.parameters.coverage >> == false ]] && echo '--no-cov' )
+                    ./scripts/check-diff ".riot/requirements/" "Changes detected after running riot. Consider deleting changed files, running scripts/compile-and-prune-test-requirements and committing the result."
+                  else
+                    echo "No changes detected, skipping test run."
+                  fi
```

Comment:
@brettlangdon do you know if we can DRY this up by putting this script somewhere in `scripts` and running that instead? Would that work?

Reply:
yeah, it _should_ work. I am game for trying it!

I am ok with that since we can try to format it a little better too rather than trying to do it as a massive oneliner.

Reply:
I've given this a try 🤞

Reply:
Attempt failed. It seems that some jobs end up with a messed up environment when running with a script, whether it's bash or sh. I've reverted the DRY changes, perhaps we can get back to this after.